sys/include/net/l2filter.h

```diff
@@ -0,0 +1,113 @@
+/*
+ * Copyright (C) 2017 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    net_l2filter Link layer address filter
+ * @ingroup     net
+ * @brief       Module for filtering (black- or white-listing) link layer
+ *              addresses
+ *
+ * This module is used as a base for filtering link layer addresses. It allows
+ * to be (globally) configured in either blacklist or in whitelist mode. This
+ * configuration is done via RIOTs module system, for whitelist mode simply
+ * include the module `L2FILTER_WHITELIST`, for blacklist mode include
+ * `L2FILTER_BLACKLIST`.
+ *
+ * The actual memory for the filter lists should be allocated for every network
+ * device. This is done centrally in netdev_t type.
+ *
+ * @{
+ * @file
+ * @brief       Link layer address filter interface definition
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ */
+
+#ifndef L2FILTER_H
+#define L2FILTER_H
+
+#include <stddef.h>
+#include <stdint.h>
+#include <stdbool.h>
+#include <errno.h>
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#ifndef L2FILTER_ADDR_MAXLEN
+/**
+ * @brief   Maximal length of addresses that can be stored in the filter list
+ */
+#define L2FILTER_ADDR_MAXLEN            (8U)
+#endif
+
+/**
+ * @brief   Number of slots in each filter list (filter entries per device)
+ */
+#ifndef L2FILTER_LISTSIZE
+#define L2FILTER_LISTSIZE               (8U)
+#endif
+
+/**
+ * @brief   Filter list entries
+ *
+ * The filter list supports address entries with differing length. This is
+ * useful e.g. for IEEE802.15.4, where the list can store short and long
+ * addresses at the same time.
+ */
+typedef struct {
+    uint8_t addr[L2FILTER_ADDR_MAXLEN];     /**< link layer address */
+    size_t addr_len;                        /**< address length in byte */
+} l2filter_t;
+
+/**
+ * @brief   Add an entry to a devices filter list
+ *
+ * @param[in] list      pointer to the filter list
+ * @param[in] addr      address to be added to list
+ * @param[in] addr_len  size of @p addr [in byte]
+ *
```

Comment:
missing `@pre`s stating the assert conditions (list and addr != NULL, and addr_len < MAX_ADDRLEN.